fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -704,6 +759,18 @@ def fixup(p: Parameter, is_sharded: bool, size: torch.Size) -> Parameter:
         del self.orig_sizes
         self._reset_lazy_init()
 
+    def parameters(self, recurse: bool = True) -> Iterator[Parameter]:
+        """Returns an iterator over the module parameters, yielding all the parameters
+        part of the model.
+        """
+        # TODO(anj): Use `copy_into_tensor` in order to provide a copy of the
+        # parameters and not the actual parameters. Ideally we don't users to operate on
+        # actual params.
+        if self.ssd_offload:
+            self.ssd_buffer.from_disk(self.buffer_size)
```

Comment:
lacking some context here (these collapsed views are killing me), but to make sure I understand, ssd_buffer is a single flat tensor which holds all of this module's parameters, right ? Noob question, but is there any value in having a mixed strategy ? (guess is no, but I'm wondering about parameters which would be shared across modules for instance)

Reply:
I don't think there is any value right now. Maybe it is something to consider in the future. Prefer to keep it simple in the beginning :)